File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task31_PERPLEXITY_llama-3-sonar-large-32k-chat.java
public class Main {
    public static void main(String[] args) {
        System.out.println(DNATranslate("ATTGC")); // --> "TAACG"
        System.out.println(DNATranslate("GTAT")); // --> "CATA"
    }

    public static String DNATranslate(String dna) {
        String complement = "";
        for (char c : dna.toCharArray()) {
            switch (c) {
                case 'A':
                    complement += 'T';
                    break;
                case 'T':
                    complement += 'A';
                    break;
                case 'C':
                    complement += 'G';
                    break;
                case 'G':
                    complement += 'C';
                    break;
            }
        }
        return complement;
    }
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task31_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <string>
#include <unordered_map>

std::string DNAStrand(const std::string& dna) {
    std::unordered_map<char, char> complement = {{'A', 'T'}, {'T', 'A'}, {'C', 'G'}, {'G', 'C'}};
    std::string result;
    for (char c : dna) {
        result += complement[c];
    }
    return result;
}